scripts/cluster_buildings.py

```diff
@@ -166,16 +170,26 @@ def get_central_points_geojson_with_buildings(
             [all_buildings_class, buildings_class], ignore_index=True
         )
 
-    # Save all the results to their respective output files
-    all_central_features.to_file(
-        output_filepath_centroids, driver="GeoJSON"
-    )  # Save cluster centroids as GeoJSON
-    all_microgrid_buildings.to_file(
-        output_filepath_buildings, driver="GeoJSON"
-    )  # Save clustered buildings as GeoJSON
-    all_buildings_class.to_csv(
-        output_path_csv, index=False
-    )  # Save building type counts as CSV
+    # Save centroids (unchanged)
+    all_central_features.to_file(output_filepath_centroids, driver="GeoJSON")
+
+    # Ensure unique columns, save buildings with GeoPandas (no manual JSON)
+    if all_microgrid_buildings.columns.duplicated().any():
+        all_microgrid_buildings = all_microgrid_buildings.loc[
+            :, ~all_microgrid_buildings.columns.duplicated(keep="first")
+        ].copy()
+
+    # Ensure WGS84 for GeoJSON output
+    gdf_out = (
+        all_microgrid_buildings.to_crs("EPSG:4326")
+        if all_microgrid_buildings.crs
+        and all_microgrid_buildings.crs.to_string().upper() != "EPSG:4326"
+        else all_microgrid_buildings
```

Comment:
This if/else shouldn't be needed. A problem may lay in the initialization of all_microgrids_buildings. you can also use gdf.set_crs(...) to avoid this